flagscale/train/hetero/parallel_context.py

```diff
@@ -242,16 +260,9 @@ def build_process_group(
             nccl_option_name = self.get_nccl_option_name(token, is_expert=is_expert)
             pg_options = get_nccl_options(nccl_option_name, self._nccl_comm_cfgs)
             ranks = self._rank_mapper.to_physical_ranks(logical_ranks)
-            group = torch.distributed.new_group(
-                ranks,
-                backend=self._distributed_backend,
-                timeout=self._timeout,
-                pg_options=pg_options,
-            )
+            group = create_group(ranks, timeout=self._timeout, backend=self._distributed_backend, pg_options=pg_options)
             if gloo:
-                group_gloo = torch.distributed.new_group(
-                    ranks, timeout=self._timeout, backend="gloo"
-                )
```

Comment:
Since Megatron-LM has the group_desc arg for each group construction, we also need a way to provide them.

Reply:
> Since Megatron-LM has the group_desc arg for each group construction, we also need a way to provide them.

Done. Use the group name as group_desc in `parallel_context`.